compiler/transforms/set_memory_layout.py

```diff
@@ -9,11 +9,98 @@
     op_type_rewrite_pattern,
 )
 
+from compiler.dialects.kernel import QMacOp, RescaleOp
 from compiler.dialects.snax import LayoutCast
 from compiler.dialects.tsl import TiledStridedLayoutAttr
 from compiler.ir.tsl import Stride, TiledStride, TiledStridedLayout
 
 
+class AddMemoryLayoutSIMD(RewritePattern):
```

Comment:
Add a docstring here please that clarifies that it only applies this in case of a SIMD operation